crates/biome_analyze/src/lib.rs

```diff
@@ -473,18 +490,16 @@ where
                 (self.emit_signal)(&signal)?;
             }
 
-            let rule = match kind {
-                SuppressionKind::Everything => None,
-                SuppressionKind::Rule(rule) => Some(rule),
-                SuppressionKind::MaybeLegacy(rule) => Some(rule),
-                SuppressionKind::Deprecated => None,
+            let (rule, value) = match kind {
+                SuppressionKind::Everything => (None, None),
+                SuppressionKind::Rule(rule) => (Some(rule), None),
+                SuppressionKind::RuleWithValue(rule, value) => (Some(rule), Some(value)),
+                SuppressionKind::MaybeLegacy(rule) => (Some(rule), None),
```

Comment:
Since you're re-using the legacy suppression comment and giving it a new semantic, maybe we should remove the name `MaybeLegacy` and change it to a new one.